### PR TITLE
SDI-78 Create un-paged version of codes for a domain

### PR DESCRIPTION
**Pull request by andymarke**
The previous url was badly named as `/domains/{domain}` and also returns a paged result that almost certainly isn't required.

New url is `/domains/{domain}/codes`  and is unpaged. Also sub-codes is never returned since in live it is *never* requested

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/repository/ReferenceDataRepository.java [hunk 1]
```diff
@@ -75,7 +75,7 @@
                 createParams("domain", domain, "code", code),
                 REF_CODE_DETAIL_ROW_MAPPER);
 
-        final var referenceCodeAsList = convertToReferenceCodes(rcdResults, false);
+        final var referenceCodeAsList = convertToReferenceCodes(rcdResults);
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/ReferenceDataRepository.java:79]: Unrelated warning clean-up; this was always false, so removed with the consequences below

File: src/main/java/uk/gov/justice/hmpps/prison/repository/ReferenceDataRepository.java [hunk 2]
```diff
@@ -205,9 +218,7 @@ public void updateReferenceCode(final String domain, final String code, final Re
         final var collectedSubCodes = collectByParentCode(subCodes);
 
         // Inject associated sub-codes into each 'parent' reference code
-        refCodes.getItems().forEach(rc -> {
-            rc.setSubCodes(collectedSubCodes.get(rc.getCode()));
-        });
+        refCodes.getItems().forEach(rc -> rc.setSubCodes(collectedSubCodes.get(rc.getCode())));
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/ReferenceDataRepository.java:222]: remove warnings

**Review by petergphillips — APPROVED** [review]
LGTM
